libs/initScripts/computeRSet_with_HCs.js

```diff
@@ -0,0 +1,355 @@
+        throw e
+      }
+    }
+  }
+
+  let userIds = [5] // Object.keys(userIdToWallet)
+  console.log(`\n${userIds.length} users have no replica sets\nThis is ${userIds.length * 100 / numOfUsers}% of user base`)
+
+  // Get all non-Audius SPs
+  // const audiusInfraSpIdsArr = [1, 2, 3, 4/*, umID*/] // TODO: uncomment later
+
+  // Compute secondaries for users while keeping UM as primary for the new replica set
+  const UMSpId = (await audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceProviderInfoFromEndpoint(USER_METADATA_ENDPOINT)).spID
+
+  const audiusInfraSpIdsArr = [UMSpId]
+  const audiusInfraSpIds = new Set(audiusInfraSpIdsArr) // when UM is registered, exclude it as secondary
+  let spIdToEndpointAndCount = {}
+
+  const sps = await audiusLibs.ethContracts.getServiceProviderList(CONTENT_NODE_TYPE)
+  sps
+    .filter(sp => !audiusInfraSpIds.has(sp.spID))
+    .forEach(sp => {
+      spIdToEndpointAndCount[sp.spID] = { endpoint: sp.endpoint, selected: 0 }
+    })
+
+  console.log('spIds', spIdToEndpointAndCount)
+
+  // Do health check for all non-Audius SPs
+  const healthCheckedSPs = await Promise.all(
+    Object.entries(spIdToEndpointAndCount)
+      .map(entry => performHealthCheck(entry))
+  )
+
+  console.log('Health check responses:', healthCheckedSPs)
+
+  // Filter out unhealthy nodes and retrieve id of each healthy node
+  const spIds = healthCheckedSPs
+    .filter(response => response.status === 200)
+    .map(response => parseInt(response.id))
+
+  console.log('\nComputing replica sets....')
+
+  let userIdToRSet = {}
+  userIds.forEach(id => {
+    // Randomly select two secondaries from spIds
+    let replicaSet = []
+    let secondary1Index = Math.floor(Math.random(0) * spIds.length)
+    let secondary2Index = -1
+    while (secondary2Index === -1 || secondary1Index === secondary2Index) {
+      secondary2Index = Math.floor(Math.random(0) * spIds.length)
+    }
+
+    replicaSet.push(spIds[secondary1Index])
+    replicaSet.push(spIds[secondary2Index])
+
+    userIdToRSet[id] = replicaSet
+
+    // Keep track of number of times the secondary was chosen
+    spIdToEndpointAndCount[spIds[secondary1Index]].selected += 1
+    spIdToEndpointAndCount[spIds[secondary2Index]].selected += 1
+  })
+
+  // Print out user-to-rset mapping
+  console.log('<userIds - replica sets>:\n', userIdToRSet)
+  // Print out all healthy, non-Audius SPs to # of times they were assigned as secondaries
+  console.log('<service providers - number of times selected as secondary>:\n', spIdToEndpointAndCount, '\n')
+
+  // Write data to file
+  writeDataToFile(spIdToEndpointAndCount, userIdToRSet)
+
+  // Trigger sync for newly selected secondaries
+  console.log(`\nSyncing across new secondaries....`)
+  let userIdToRSetArr = Object.entries(userIdToRSet)
+  for (i = 0; i < userIds.length; i++) {
+    const userId = parseInt(userIdToRSetArr[i][0])
+    const replicaSetSPIds = userIdToRSetArr[i][1]
```

Comment:
is this just secondaries? if so let's explicitly call this 'replicaSetSecondarySpIds'